### DISCUSSION TITLE
How to convert a white icon PNG file to SVG?

### DISCUSSION THREAD

**Discussion by cray12399** [ImageMagick/ImageMagick#3475] [Help]
I am trying to convert a white PNG icon to an SVG file. I am running the command:
`convert firefox-512.png firefox.svg
`
However, when I run it, the white parts of the image are made transparent and the transparent parts are made black. How can I run it so that the white icon and transparent background are preserved in the SVG? I tried using the transparent-color and transparent flags, but no matter where I place them in the command, they do absolutely nothing.

**Comment by fmw42** [comment]
It would help to have your PNG and also to know your IM version and platform/OS!

**Reply by cray12399** [reply]
My bad. Should've included those! My IM version is ImageMagick 6.9.11-27 Q16 x86_64 2020-08-11 https://imagemagick.org and I am on Fedora Linux 33 Kernel 5.11.
PNG is attached.
![firefox-512](https://user-images.githubusercontent.com/66784543/113061558-3d95c900-9180-11eb-90f3-be5d5410a7e3.png)
It appears white because it is a white icon on a transparent background

**Comment by snibgo** [comment]
ImageMagick is a raster image processor. It won't convert raster images to vector images. For that, you need a tool like potrace.

The best that IM can do is create an SVG file with embedded raster image.

**Reply by fmw42** [reply]
Correct me if I am wrong, but I thought ImageMagick would convert the rasterized image to vector if one had potrace or perhaps autotrace installed. It may just be vectorized pixels though.

**Reply by cray12399** [reply]
I have potrace installed and tried it, but there are zero docs on it so I have no idea how to use it to convert the icon to an SVG. Potrace actually does the exact same thing ImageMagick does when I try to convert. Would you happen to know how I can convert my PNG to an SVG using potrace?

**Comment by snibgo** [comment]
I have potrace installed, on the system path, but IM doesn't call it.

Run the IM command with "-verbose" to see if it is calling potrace as a delegate.

Run "-potrace --help" for help on potrace options. For example:
```
magick  113061558-3d95c900-9180-11eb-90f3-be5d5410a7e3.png x.pbm

potrace -b svg -o x.svg x.pbm
```
That creates x.svg:
```
<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="512.000000pt" height="512.000000pt" viewBox="0 0 512.000000 512.000000"
 preserveAspectRatio="xMidYMid meet">
<metadata>
Created by potrace 1.15, written by Peter Selinger 2001-2017
</metadata>
<g transform="translate(0.000000,512.000000) scale(0.100000,-0.100000)"
fill="#000000" stroke="none">
<path d="M0 2560 l0 -2560 2560 0 2560 0 0 2560 0 2560 -2560 0 -2560 0 0
-2560z m2869 2430 c323 -34 670 -158 971 -347 52 -33 145 -86 205 -118 230
-122 369 -237 493 -407 35 -47 103 -171 125 -228 11 -27 -5 82 -29 204 -28
141 -28 137 -8 129 42 -16 194 -296 252 -463 60 -173 110 -430 128 -665 l7
-90 14 39 c7 22 18 68 25 103 12 65 30 81 43 41 4 -13 10 -132 12 -264 6 -274
0 -331 -69 -644 -62 -288 -128 -467 -270 -742 -145 -280 -292 -489 -486 -689
-365 -379 -760 -608 -1217 -705 -109 -24 -147 -26 -426 -31 -428 -7 -604 13
-879 102 -389 126 -681 308 -999 619 -286 282 -461 548 -596 911 -89 239 -120
421 -119 705 0 224 14 330 64 498 22 74 24 90 11 79 -9 -7 -33 -27 -54 -45
-21 -18 -42 -32 -47 -32 -14 0 -12 116 2 203 9 56 29 107 90 228 43 86 94 174
113 197 49 57 55 79 66 247 16 267 54 451 123 599 42 91 121 198 143 194 11
-2 24 -35 43 -113 32 -126 62 -212 98 -273 l24 -43 122 119 c147 143 243 221
396 320 279 183 674 329 980 361 109 12 540 12 649 1z"/>
<path d="M2287 4765 c-467 -68 -880 -264 -1202 -569 l-80 -76 45 8 c74 12 321
8 359 -5 32 -12 35 -11 105 46 135 111 304 201 456 244 126 36 160 41 160 24
0 -8 -17 -27 -37 -43 -72 -55 -186 -169 -223 -224 -41 -61 -84 -161 -102 -241
l-12 -52 109 -104 c61 -56 127 -115 147 -129 46 -33 122 -51 247 -60 118 -8
131 -14 131 -60 0 -79 -32 -114 -245 -273 -66 -49 -133 -103 -149 -120 l-30
-31 19 -37 c36 -75 56 -160 62 -260 5 -86 3 -103 -9 -108 -9 -4 -86 8 -172 24
-183 36 -210 38 -254 11 -35 -22 -71 -87 -72 -130 0 -45 83 -145 214 -261 165
-144 299 -192 446 -158 25 5 97 34 161 64 177 83 269 100 384 70 160 -42 295
-154 295 -245 0 -57 -22 -70 -114 -70 l-77 0 -137 -88 c-216 -139 -252 -159
-348 -191 -123 -41 -246 -61 -382 -61 l-115 0 19 -21 c10 -12 63 -55 117 -96
167 -125 391 -223 579 -253 113 -18 355 -8 434 18 249 83 463 245 679 517 90
113 110 133 118 119 12 -21 1 -185 -18 -274 -7 -36 -12 -66 -10 -68 1 -2 23
30 48 70 87 138 152 304 184 466 14 70 17 142 17 370 -1 253 1 283 15 280 15
-3 101 -160 132 -243 l16 -40 18 75 c23 98 32 370 16 511 -7 57 -25 170 -42
252 -16 81 -29 152 -29 157 0 31 93 -13 160 -76 l48 -44 -5 47 c-9 95 -25 155
-63 231 -111 221 -368 421 -775 603 -209 94 -206 108 18 118 l138 6 -36 43
c-71 82 -190 150 -363 207 -62 21 -110 41 -107 46 7 10 -33 21 -191 53 -220
45 -509 58 -697 31z"/>
</g>
</svg>
```
... which looks reasonable. Use potrace option "--fillcolor" etc if you want.
